### PR TITLE
Build stratisd with the udev sync devicemapper-rs branch

### PR DESCRIPTION
**Pull request by bmr-cymru**
Update the build dependencies for devicemapper-rs to reference the bmr-udev-sync branch and include `ilbdevmapper.h` in the dependencies for CI environments.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -153,7 +156,7 @@ jobs:
           --print-address
       - name: Run stratis-cli tests
         run: >
-          RUST_LOG=stratisd=debug
+          RUST_LOG=trace
```
Comment (mulkieran) [inline .github/workflows/main.yml:159]: For stratis-cli tests, might as well stick with DEBUG level. They are run against the sim engine, and so don't tend to cause udev events of interest.

Comment (bmr-cymru) [reply .github/workflows/main.yml:159]: Thanks - I will make that change on my branch, and disable the other tests that I'm not concerned with to keep the runs to a minimum.

**Comment by bmr-cymru** [comment]
The CI failures from Friday are due to missing udev rules / support programs in the test runner. This is due to the stratis-ci-container-action using the old "build-and-install" target that was removed in January. I've pushed an update to the GitHub action and modified this branch to point to it (stratis-ci-container-action@v2).

**Comment by bmr-cymru** [comment]
Rebased & tidied up

**Review by mulkieran — APPROVED** [review]

**Review by jbaublitz — COMMENTED** [review]
Could you explain why we need the remaining udev_settle calls? I was under the impression that this would would make those unnecessary.

**Comment by mulkieran** [comment]
@bmr-cymru Ok! New devicemapper-rs release is up; so you can fix up Cargo.toml and put it up for final review.

**Review by jbaublitz — APPROVED** [review]

**Comment by mulkieran** [comment]
@bmr-cymru I think we're all good with this. Plz squash any commits if you want to and rebase, and I'll merge.

**Comment by mulkieran** [comment]
Oh, I see that this was actually due to https://bugzilla.redhat.com/show_bug.cgi?id=2176263 . Thanks all for identifying that.

**Comment by mulkieran** [comment]
We have to hold this a bit to do some spec file testing.

**Pull request merged**